### PR TITLE
Update scripts: Force bash & Handle go 1.18 go get deprecation

### PR DESCRIPTION
**Pull request by juanig1**
This pr:
Handles deprecation of go get (based on this [kubebuilder pr](https://github.com/kubernetes-sigs/kubebuilder/pull/2486))
Handles cases where bash is not the default non-interactive shell (e.g., Debian).

Note:
Swapping 'source' for '.' and revisiting [line 5](https://github.com/juanig1/artillery-operator/blob/a8b92de72c081feb34d156ef799f69354d6e712e/hack/kind/kind-with-registry.sh#L5) of hack/kind/kind-with-registry.sh (bad substitution error) may be an option if we want to keep using /bin/sh.

### CODE REVIEW

**Comment by juanig1** [comment]
This is a workaround for my system; but may be the wrong approach. Feedback is appreciated!

**Review by ezodude — CHANGES_REQUESTED** [review]
All in all makes sense. 👍 

Can we update the `goreleaser` [Makefile task](https://github.com/artilleryio/artillery-operator/blob/main/Makefile#L260) to work like the other commands using go install, i.e. Kustomize, etc.. (it already does that but we no longer need the `go-get-install` function)

We'll need to revisit this once the latest Kubebuilder/Operator-SDK is released.

**Review by ezodude — APPROVED** [review]
LGTM 💯

File: hack/prom-pushgateway/up.sh [hunk 1]
```diff
@@ -34,7 +34,7 @@ if [ ! -f "${kube_prom_zip}" ]; then
 fi
 
 # unpack the operator
-tar -zxvf "${kube_prom_zip}" -C "${script_dir}"
+unzip "${kube_prom_zip}" -d "${script_dir}"
```
Comment (ezodude) [inline hack/prom-pushgateway/up.sh:37]: Turns out on a Mac, `tar` symlinks to `bsdtar` not GNU tar. There are some minor differences there, so we're definitely not using the same tar.

`unzip` is a good replacement. 👍

**Review by ezodude — APPROVED** [review]
👍 lgtm

**Pull request merged**